packages/naming.cell.match/cell-match.js

```diff
@@ -0,0 +1,102 @@
+'use strict';
+
+const assert = require('assert');
+
+const BemCell = require('@bem/sdk.cell');
+const bemNamingParse = require('@bem/sdk.naming.entity.parse');
+const pathPatternParser = require('@bem/sdk.naming.cell.pattern-parser');
+
+const ALPHANUM_RE = '[a-z][\\w\\-]*';
+const resc = s => String(s).replace(/[\\^$*+?.()|[\]{}]/g, '\\$&');
+
+const buildPathParseMethod = ({ fs: { pattern, delims: dd = {}, scheme = 'nested' }, delims, wordPattern: wp = ALPHANUM_RE }) => {
+    const separation = pathPatternParser(pattern);
+
+    const dElem = 'elem' in dd ? dd.elem : (delims.elem || '__');
+    const dMod = 'mod' in dd ? dd.mod : (Object(delims.mod).name || (typeof delims.mod === 'string' && delims.mod) || '_');
+
+    const entityRe = ({
+        flat: [`(?:(${wp})(?:/(${ALPHANUM_RE})`, ')?)?', (entity, { dir }) => (entity.block === dir)],
+        nested: [`(?:(${wp}(?:|/${dElem}${wp})?(?:|/${dMod}${wp})?)(?:/(${ALPHANUM_RE})`, ')?)?', (entity, { dir }) => {
```

Comment:
Too complex for me. Can we write here a helper? Now I need to take a pencil/paper and start count braces

Reply:
Why you should count it? Who or what forces you to do it?

Reply:
Guess we need to move it out to separate package along with scheme path builders.

Probably it should be #304 for a while.